### ISSUE TITLE
Use DOMPurify to sanitized output from marked?

### ISSUE THREAD

**Issue by hansonw** [atom/notifications#185]
### Description

Currently, the notifications package takes the raw output of `marked` and writes it to the notification's `innerHTML`. If notification content can be user-controlled in some way, this makes it easy to write security exploits using tags like `<iframe></iframe>` (despite Atom's cross-origin protections, it's possible to load arbitrary HTML files on disk, for example).

### Steps to Reproduce

Example:

```
atom.notifications.addInfo('<iframe src="file:///etc/passwd" />')
```

While one could argue that it's the responsibility of the notifications caller to sanitize the HTML before sending it through, it's often not obvious that 1) notifications take Markdown and 2) notifications accept HTML in Markdown.

Having HTML in notifications is definitely still useful - but would it be an acceptable compromise to run the output of `marked` through https://github.com/cure53/DOMPurify before displaying it? (This is what settings-view does as well: https://github.com/atom/settings-view/blob/e962a54f488cfd5b3b0be5ab99be70fa8a2260e8/lib/package-readme-view.js#L43)

I can submit a pull request if this seems reasonable.

**Comment by lee-dohm** [comment]
@hansonw Yeah, we would totally :heart: a PR for this. Let me know if you need any information.
